Fatal Python error: ffi.from_handle() detected that the address passed points to garbage

Python tests are hanging on TOSS 3 system:
```
PASS: ../t/t9990-python-tests.t 1 test.barrier.TestBarrier.test_eight
PASS: ../t/t9990-python-tests.t 2 test.barrier.TestBarrier.test_single
PASS: ../t/t9990-python-tests.t 3 test.event.TestEvent.test_full_event
PASS: ../t/t9990-python-tests.t 4 test.event.TestEvent.test_t1_0_sub
PASS: ../t/t9990-python-tests.t 5 test.event.TestEvent.test_t1_1_unsub
PASS: ../t/t9990-python-tests.t 6 test.handle.TestHandle.test_create_handle
PASS: ../t/t9990-python-tests.t 7 test.handle.TestHandle.test_log
PASS: ../t/t9990-python-tests.t 8 test.handle.TestHandle.test_rpc_ping
PASS: ../t/t9990-python-tests.t 9 test.json_t.TestJsonCInterface.test_create_empty
PASS: ../t/t9990-python-tests.t 10 test.json_t.TestJsonCInterface.test_with_data
PASS: ../t/t9990-python-tests.t 11 test.kvs.TestKVS.test_bad_init
PASS: ../t/t9990-python-tests.t 12 test.kvs.TestKVS.test_exists_dir
PASS: ../t/t9990-python-tests.t 13 test.kvs.TestKVS.test_exists_false
PASS: ../t/t9990-python-tests.t 14 test.kvs.TestKVS.test_exists_true
PASS: ../t/t9990-python-tests.t 15 test.kvs.TestKVS.test_fill
PASS: ../t/t9990-python-tests.t 16 test.kvs.TestKVS.test_iterator
PASS: ../t/t9990-python-tests.t 17 test.kvs.TestKVS.test_key_at
PASS: ../t/t9990-python-tests.t 18 test.kvs.TestKVS.test_kvs_dir_open
PASS: ../t/t9990-python-tests.t 19 test.kvs.TestKVS.test_read_non_existent
PASS: ../t/t9990-python-tests.t 20 test.kvs.TestKVS.test_read_non_existent_basedir
PASS: ../t/t9990-python-tests.t 21 test.kvs.TestKVS.test_remove
PASS: ../t/t9990-python-tests.t 22 test.kvs.TestKVS.test_set_deep
PASS: ../t/t9990-python-tests.t 23 test.kvs.TestKVS.test_set_dict
PASS: ../t/t9990-python-tests.t 24 test.kvs.TestKVS.test_set_float
PASS: ../t/t9990-python-tests.t 25 test.kvs.TestKVS.test_set_int
PASS: ../t/t9990-python-tests.t 26 test.kvs.TestKVS.test_set_list
PASS: ../t/t9990-python-tests.t 27 test.kvs.TestKVS.test_set_none
PASS: ../t/t9990-python-tests.t 28 test.kvs.TestKVS.test_set_string
PASS: ../t/t9990-python-tests.t 29 test.kvs.TestKVS.test_walk
PASS: ../t/t9990-python-tests.t 30 test.kvs.TestKVS.test_walk_with_no_handle
PASS: ../t/t9990-python-tests.t 31 test.request.TestRequestMethods.test_no_topic_invalid
PASS: ../t/t9990-python-tests.t 32 test.request.TestRequestMethods.test_null_payload
PASS: ../t/t9990-python-tests.t 33 test.watchers.TestTimer.test_s1_0_timer_add
PASS: ../t/t9990-python-tests.t 34 test.watchers.TestTimer.test_s1_1_timer_remove
PASS: ../t/t9990-python-tests.t 35 test.watchers.TestTimer.test_timer_add_negative
```
after waiting many minutes, I hit ctrl-C
```
^Cmake[3]: *** Deleting file `../t/t9990-python-tests.log'tap-driver.sh: fatal: I/O or internal error
make[3]: *** [../t/t9990-python-tests.log] Error 1
make[2]: *** [check-TESTS] Interrupt
make[1]: *** [check-am] Interrupt
make: *** [check-recursive] Interrupt
```

This seems  to reproduce (three for three tries at this point).   This is on current master `0.2.0-229-gd1ddd8d`

On my desktop the next test would be
```
PASS: t9990-python-tests.t 34 test.watchers.TestTimer.test_timer_with_reactor
```
(Note tests appear to run in a different order there - barrier tests show up last not first)

--- a/src/bindings/python/flux/core/watchers.py
+++ b/src/bindings/python/flux/core/watchers.py
@@ -57,7 +57,7 @@ class MessageWatcher(Watcher):
         self.fh = flux_handle
         self.cb = callback
         self.args = args
-        wargs = ffi.new_handle(self)
+        self.wargs = ffi.new_handle(self)
         c_topic_glob = ffi.new('char[]', topic_glob)
         match = ffi.new('struct flux_match *', {
             'typemask': type_mask,
@@ -67,7 +67,7 @@ class MessageWatcher(Watcher):
         })
         self.handle = raw.flux_msg_handler_create(self.fh.handle, match[0],
                                                   message_handler_wrapper,
-                                                  wargs)
+                                                  self.wargs)
 
     def start(self):
         raw.flux_msg_handler_start(self.handle)
@@ -95,10 +95,10 @@ class TimerWatcher(Watcher):
         self.cb = callback
         self.args = args
         self.handle = None
-        wargs = ffi.new_handle(self)
+        self.wargs = ffi.new_handle(self)
         self.handle = raw.flux_timer_watcher_create(
             raw.flux_get_reactor(flux_handle),
-            float(after), float(repeat), timeout_handler_wrapper, wargs)
+            float(after), float(repeat), timeout_handler_wrapper, self.wargs)
 
 @ffi.callback('flux_watcher_f')
 def fd_handler_wrapper(handle_trash, fd_watcher_s, revents,
@@ -116,8 +116,8 @@ class FDWatcher(Watcher):
         self.cb = callback
         self.args = args
         self.handle = None
-        wargs = ffi.new_handle(self)
+        self.wargs = ffi.new_handle(self)
         self.handle = raw.flux_fd_watcher_create(
             raw.flux_get_reactor(flux_handle),
-            self.fd_int, self.events, fd_handler_wrapper, wargs)
+            self.fd_int, self.events, fd_handler_wrapper, self.wargs)
 

